dotnet/src/SemanticKernel/Orchestration/CancellationTokenExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Orchestration;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the namespace of CancellationToken
+namespace System.Threading;
+#pragma warning restore IDE0130
+
+internal static class CancellationTokenExtensions
+{
+    /// <summary>
+    /// Combine two cancellation tokens, reusing the given tokens
+    /// unless they are both not null/default.
+    /// </summary>
+    /// <param name="token">Cancellation token 1</param>
+    /// <param name="secondToken">Cancellation token 2</param>
+    /// <returns>A combination of the two tokens</returns>
+    internal static CombinedCancellationToken CombineWith(
+        this CancellationToken token, CancellationToken secondToken)
+    {
+        return new CombinedCancellationToken(token, secondToken);
+    }
```

Comment:
I don't think this is used either, and can probably be removed, but if it is, this appears exactly equivalent to:

```suggestion
    internal static CancellationTokenSource CombineWith(
        this CancellationToken token, CancellationToken secondToken)
    {
        return CancellationTokenSource.CreateLinkedTokenSource(token1, token2);
    }
```
...both return an IDisposable object with a `.Token` property.